conda_forge_tick_action/automerge.py

```diff
@@ -139,14 +142,62 @@ def _check_github_statuses(statuses, extra_ignored_statuses=None):
             return True
 
 
+# https://stackoverflow.com/questions/6194499/pushd-through-os-system
+@contextlib.contextmanager
+def pushd(new_dir):
+    previous_dir = os.getcwd()
+    os.chdir(new_dir)
+    try:
+        yield
+    finally:
+        os.chdir(previous_dir)
+
+
+def _get_curr_maintainers(feedstock):
+    """get the current maintainers.
+
+    We always pull from master upstream. This important since the list in the
+    branch could have been changed to add new maintainers not yet approved by the
+    current ones.
+    """
+
+    assert feedstock.endswith("-feedstock"), (
+        "The repo '%s' is not a valid feedstock for getting maintainers!" % feedstock
+    )
+    url = "https://github.com/conda-forge/%s.git" % feedstock
+    with tempfile.TemporaryDirectory() as tmpdir:
+        with pushd(tmpdir):
+            subprocess.run(
+                ["git", "clone", url],
+                check=True
+            )
+            assert os.path.exists(os.path.join(feedstock, "recipe", "meta.yaml")), (
+                "Recipe did not clone correctly when getting maintainers!"
+            )
+            with open(os.path.join(feedstock, "recipe", "meta.yaml"), "r") as fp:
+                keep_lines = []
+                skip = True
+                for line in fp.readlines():
+                    if line.startswith("extra:"):
+                        skip = False
+                    if not skip:
+                        keep_lines.append(line)
+    yaml = YAML()
+    meta = yaml.load("".join(keep_lines))
+    return meta.get("extra", {}).get("recipe-maintainers", [])
```

Comment:
This doesn't take teams into account.

Reply:
Ah yes. Will fix.

Reply:
actually no, we can't fix this. We don't remove people from teams currently so this would allow any maintainer ever, not just the current ones.